server/server.py

```diff
@@ -36,33 +30,10 @@ def authorized(resource, action):
     def wrapped(f):
         @functools.wraps(f)
         def wrapper(*args, **kwargs):
-            iam_url = IAM_SERVER + "/allowed"
-            body = json.dumps({
-                "resource": resource,
-                "action": action,
-            }).encode("utf-8")
-            headers = {
-                "Authorization": request.headers.get("Authorization", None),
-                "Origin": API_AUDIENCE,
-            }
-            r = urllib.request.Request(iam_url, data=body, headers=headers)
-            try:
-                resp = urllib.request.urlopen(r)
-            except urllib.error.HTTPError as e:
-                raise AuthError(e.read().decode("utf-8"), e.code)
-
-            payload = json.loads(resp.read().decode("utf-8"))
-
-            if not payload["allowed"]:
-                raise AuthError({
-                    "code": "not_allowed",
-                    "description": "This JWT Access Token is not authorized."
-                }, 403)
-
+            jwt = request.headers.get("Authorization", None)
+            payload = allowed(IAM_SERVER, API_AUDIENCE, resource=resource, action=action, jwt=jwt)
```

Comment:
We could define a global currified allowed:

`allowed = functools.partial(IAM_SERVER, API_AUDIENCE....)`